plugin-server/src/main/utils.ts

```diff
@@ -35,3 +42,10 @@ export async function runInstrumentedFunction<T, E>({
         clearTimeout(timeout)
     }
 }
+
+const instrumentedFunctionDuration = new Histogram({
+    name: 'instrumented_function_duration_seconds',
+    help: 'Processing time and success status of internal functions',
+    labelNames: ['function', 'success'],
+    buckets: exponentialBuckets(0.025, 2, 12), // Saturates at 51 seconds
```

Comment:
Is this based on the range of existing statsd data?

12 buckets sounds like a lot of granularity. I guess that's fine though.

Reply:
Agreed, let me check the range of values. IIRC we wrap the app calls with these, so values can be as high as 30 seconds.
I though about having the success rate in a separate counter, to avoid doubling the buckets, but I think it's useful information (if the processing time goes down, are we still succeeding there or failing-fast?). WDYT?

Reply:
Let's go with what you've got here. Feels like it answers useful questions, it's useful to have both the success and failure characteristics and the aggregate easily to hand.

Reply:
Looking at today's trends for the existing`plugin_server_kafka_queue_each_batch_ingestion` metric
![image](https://user-images.githubusercontent.com/6241083/215767569-968aa3fe-8125-42f6-b923-29f2cbf25cd2.png), we're covering a range from 16ms to 50+ seconds, consistent with the range I'm targetting here. We could use a factor of 4 initially to use less buckets, and increase it if needed?

Reply:
Yeah I feel like 6 buckets would be enough here and like you say we can increase if that turns out not to be the case.

Reply:
Also, what a range!